src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets

```diff
@@ -322,15 +341,24 @@
         <Configuration>$(ConfigurationGroup)$(ConfigurationSuffix)</Configuration>
         <BuildNumber>$(BuildMoniker)</BuildNumber>
         <Branch>$(Branch)</Branch>
-      </BuildComplete>
+      </BuildCompleteTemplate>
+      <BuildCompleteTemplateV2 Include="%(TestListFile.BuildCompleteJson)">
+        <DropContainerSAS>$(DropUriReadOnlyToken)</DropContainerSAS>
+        <ListUri>$(DropUri)%(TestListFile.Filename)%(TestListFile.Extension)$(DropUriReadOnlyToken)</ListUri>
+        <QueueId>$(TargetQueue)</QueueId>
+        <ResultsUri>$(ResultsUri)/%(TestListFile.CorrelationId)</ResultsUri>
+        <ResultsUriRSAS>$(ResultsReadOnlyToken)</ResultsUriRSAS>
+        <ResultsUriWSAS>$(ResultsWriteOnlyToken)</ResultsUriWSAS>
+        <Build>$(CurrentDate)</Build>
+        <Build Condition="'$(IsOfficial)'=='true'">$(OfficialBuildId)</Build>
+        <Type>$(HelixJobType)</Type>
+        <Source>$(HelixSource)</Source>
+        <Properties>{ &quot;architecture&quot; : &quot;$(Platform)&quot;, &quot;configuration&quot;: &quot;$(ConfigurationGroup)&quot;, &quot;operatingSystem&quot; : &quot;$(TargetQueue)&quot; }</Properties>
+      </BuildCompleteTemplateV2>
+      <BuildComplete Condition="'$(_UseUpdatedHelixApi)'=='true'" Include="@(BuildCompleteTemplateV2)"/>
+      <BuildComplete Condition="'@(BuildComplete)'==''" Include="@(BuildCompleteTemplate)"/>
```

Comment:
More reason to move the BuildComplete stuff to its own individual files, then you can import those files based on "$(_UseUpdatedHelixApi)" and remove the conditionals on this property.